src/x509.c

```diff
@@ -5195,840 +5195,1138 @@ int wolfSSL_X509_cmp(const WOLFSSL_X509 *a, const WOLFSSL_X509 *b)
             }
-            if (x509->notAfter.length > 0) {
-                if (GetTimeString(x509->notAfter.data, ASN_UTC_TIME,
-                    tmp, sizeof(tmp)) != WOLFSSL_SUCCESS) {
-                    if (GetTimeString(x509->notAfter.data, ASN_GENERALIZED_TIME,
-                        tmp, sizeof(tmp)) != WOLFSSL_SUCCESS) {
-                        WOLFSSL_MSG("Error getting not after date");
-                        return WOLFSSL_FAILURE;
-                    }
-                }
-            }
-            else {
-                XSTRNCPY(tmp, "Not Set", sizeof(tmp)-1);
+            tmp[0] = '\0';
+            for (i = 0; i < sz - 1 && (3 * i) < tmpSz - valSz; i++) {
+                XSNPRINTF(val, sizeof(val) - 1, "%02x:", serial[i]);
+                val[3] = '\0'; /* make sure is null terminated */
+                XSTRNCAT(tmp, val, valSz);
             }
-            tmp[sizeof(tmp) - 1] = '\0'; /* make sure null terminated */
+            XSNPRINTF(val, sizeof(val) - 1, "%02x\n", serial[i]);
+            val[3] = '\0'; /* make sure is null terminated */
+            XSTRNCAT(tmp, val, valSz);
             if (wolfSSL_BIO_write(bio, tmp, (int)XSTRLEN(tmp)) <= 0) {
                 return WOLFSSL_FAILURE;
             }
         }
-    #endif
+    }
 
-        /* print subject */
-        {
-            char* subject;
-        #ifdef WOLFSSL_SMALL_STACK
-            char* buff  = NULL;
-            int   subSz = 0;
-        #else
-            char buff[256];
-            int  subSz = 256;
-        #endif
+    /* print signature algo */
+    {
+        int oid;
+        const char* sig;
 
-        #if defined(WOLFSSL_QT)
-            subject = wolfSSL_X509_get_name_oneline(
-                              wolfSSL_X509_get_subject_name(x509), buff, subSz);
-        #else
-            subject = wolfSSL_X509_NAME_oneline(
-                              wolfSSL_X509_get_subject_name(x509), buff, subSz);
-        #endif
+        if ((oid = wolfSSL_X509_get_signature_type(x509)) <= 0) {
+            WOLFSSL_MSG("Error getting x509 signature type");
+            return WOLFSSL_FAILURE;
+        }
+        if (wolfSSL_BIO_write(bio, "        Signature Algorithm: ",
+                      (int)XSTRLEN("        Signature Algorithm: ")) <= 0) {
+            return WOLFSSL_FAILURE;
+        }
+        sig = GetSigName(oid);
+        if (wolfSSL_BIO_write(bio, sig, (int)XSTRLEN(sig)) <= 0) {
+            return WOLFSSL_FAILURE;
+        }
+        if (wolfSSL_BIO_write(bio, "\n", (int)XSTRLEN("\n")) <= 0) {
+            return WOLFSSL_FAILURE;
+        }
+    }
 
-            if (wolfSSL_BIO_write(bio, "\n        Subject: ",
-                          (int)XSTRLEN("\n        Subject: ")) <= 0) {
+    /* print issuer */
+    {
+        char* issuer;
+    #ifdef WOLFSSL_SMALL_STACK
+        char* buff  = NULL;
+        int   issSz = 0;
+    #else
+        char buff[256];
+        int  issSz = 256;
+    #endif
+
+    #if defined(WOLFSSL_QT)
```

Comment:
Why just QT here? 
The `wolfSSL_X509_get_name_oneline` is enabled with `#if defined(OPENSSL_EXTRA) && defined(XSNPRINTF)`.
I realize this is keeping existing code, but curious why the different just for QT.

Reply:
Will look into it. May be outdated in which case I'll update the guard.

Reply:
Yeah, seems outdated. This entire function is already gated on OPENSSL_EXTRA and XSNPRINTF being defined. I've removed the QT gate for printing the issuer and subject.